reconcile/external_resources/manager.py

```diff
@@ -88,6 +88,64 @@ class ReconcileAction(StrEnum):
     DESTROY_ERROR = "Resource status in ERROR state"
 
 
+class ReconciliationStatus:
+    needs_secret_sync: bool = False
+    needs_deletion: bool = False
+
+    reconcile_time: int = 0
+    resource_status: ResourceStatus
```

Comment:
note these vars are class vars, but usages are instance vars, should either use `@dataclass` or pydantic model, or init them in `__init__`